benchmark/lib/benchmark.dart

```diff
@@ -0,0 +1,84 @@
+import 'dart:io';
+
+import 'package:objectbox/objectbox.dart';
+
+import 'model.dart';
+import 'objectbox.g.dart';
+
+class Executor {
+  final Store store;
+
+  /*late final*/
+  Box<TestEntity> box;
+
+  /// list of runtimes indexed by function name
+  final times = <String, List<Duration>>{};
+
+  Executor(Directory dbDir)
+      : store = Store(getObjectBoxModel(), directory: dbDir.path) {
+    box = Box<TestEntity>(store);
+  }
+
+  void close() => store.close();
+
+  R _track<R>(String fnName, R Function() fn) {
+    final watch = Stopwatch();
+
+    watch.start();
+    final result = fn();
+    watch.stop();
+
+    times[fnName] ??= <Duration>[];
+    times[fnName].add(watch.elapsed);
+    return result;
+  }
+
+  void _print(List<dynamic> varArgs) {
+    print(varArgs.join('\t'));
+  }
+
+  void printTimes([List<String> functions]) {
+    functions ??= times.keys.toList();
+
+    // print the whole data as a table
+    _print(['Function', 'Runs', 'Average ms', 'All times']);
+    for (final fun in functions) {
+      final fnTimes = times[fun];
+
+      final sum = fnTimes.map((d) => d.inMicroseconds).reduce((v, e) => v + e);
+      final avg = sum.toDouble() / fnTimes.length.toDouble() / 1000;
+      final timesCols = fnTimes.map((d) => d.inMicroseconds.toDouble() / 1000);
+      _print([fun, fnTimes.length, avg, ...timesCols]);
+    }
+  }
+
+  List<TestEntity> prepareData(int count) {
```

Comment:
A little confused why `prepareData` and the operations (`putMany`, `updateAll`) are in this file and not e.g. `main.dart`. Should the measuring and the actual operations be separated, e.g. to allow easy re-use in the future?

Reply:
Actually, I wanted to have as much in the "lib" so that this can be also used to benchmark from a flutter application, without much code duplication.

Reply:
I'll leave this discussion open and merge anyway, we can address directly on main later if needed.

Reply:
It's fine, just wanted to point it out. Can refactor when the time comes.